pkg/server/pubsub.go

```diff
@@ -15,12 +15,22 @@
 package server
 
 import (
+	"bytes"
 	"context"
 	"fmt"
+	"io"
 
 	"cloud.google.com/go/pubsub"
 )
 
+// Detailed size limit can be found here:
+// https://cloud.google.com/pubsub/quotas#resource_limits.
+const (
+	TopicDataByteSize      = 10_000_000
+	TopicAttrKeyByteSize   = 256
+	TopicAttrValueByteSize = 1024
```

Comment:
Nit: it would be better to name this `MaxTopicDataBytes`. Usually the units are the last part of the constant name.